agent/installer/internal/algo/algo_test.go

```diff
@@ -43,4 +46,38 @@ var _ = Describe("Installer Algo Tests", func() {
 			Expect(outputBuilderCounter.LogCalledCnt).Should(Equal(stepsNum))
 		})
 	})
+	Context("When error occurs during installation", func() {
+		var (
+			err        error
+			mockUbuntu MockUbuntuWithError
+		)
+		const (
+			stepWhenError    = 5
+			obcExpectedTotal = stepWhenError*4 + 2
+		)
+		BeforeEach(func() {
+			mockUbuntu = MockUbuntuWithError{}
+			mockUbuntu.OutputBuilder = &outputBuilderCounter
+			mockUbuntu.BundlePath, err = os.MkdirTemp("", "algoErrTest")
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			installer = &BaseK8sInstaller{
+				BundlePath:      mockUbuntu.BundlePath,
+				K8sStepProvider: &mockUbuntu,
+				OutputBuilder:   &outputBuilderCounter}
+		})
+		AfterEach(func() {
+			err = os.RemoveAll(mockUbuntu.BundlePath)
```

Comment:
What we're trying to test is an abstract algorithm which does not include concrete steps.
We should create/worry about directories files.